src/utils/url_encode.rs

```diff
@@ -0,0 +1,96 @@
+use anyhow::{Error, Result};
+use percent_encoding::{
+    percent_decode, percent_encode, utf8_percent_encode, AsciiSet, NON_ALPHANUMERIC,
+};
+use std::os::unix::ffi::OsStrExt;
+use std::path::PathBuf;
+use std::str::FromStr;
+
+pub const PERCENT_ENCODE_SET: &AsciiSet = &NON_ALPHANUMERIC
+    .remove(b'-')
+    .remove(b'_')
+    .remove(b'.')
+    .remove(b'~');
+
+pub fn encode_uri(file_path: &PathBuf) -> String {
```

Comment:
Used the implementation suggested [here](https://github.com/EstebanBorai/http-server/pull/164#discussion_r926341720)